flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/common/CommonPhysicalLookupJoin.scala

```diff
@@ -98,11 +98,11 @@ abstract class CommonPhysicalLookupJoin(
     // all potential index keys, mapping from field index in table source to LookupKey
     analyzeLookupKeys(cluster.getRexBuilder, joinKeyPairs, calcOnTemporalTable)
   }
-  // remaining condition used to filter the joined records (left input record X lookup-ed records)
-  val remainingCondition: Option[RexNode] = getRemainingJoinCondition(
+  // split remaining condition into pre-filter(used to filter the left input before lookup) and
+  // remaining parts(used to filter the joined records)
+  val (finalPreFilterCondition, finalRemainingCondition) = splitRemainingJoinCondition(
```

Comment:
Maybe we can rename this method. `splitRemainingJoinCondition` looks like split the `finalRemainingCondition` instead of split condition into pre-filter and remaining.